src/services/app-config.service.js

```diff
@@ -81,3 +84,107 @@ export function deleteApp() {
     }
   );
 }
+
+export async function fetchAndStoreAppConfig() {
+  const apps = await fetchAllApps();
+  if (apps.data.length) {
+    store.dispatch("updateAppConfigurationStatus", true);
+    const currentApp = apps.data[0];
+    const appId = currentApp.ID;
+    store.dispatch("updateAppName", currentApp.name);
+    store.dispatch("updateAppId", currentApp.ID);
+
+    // Get Address
+    const appAddress = await getAddress(currentApp.address);
+    store.dispatch("updateSmartContractAddress", appAddress);
+
+    const env = store.getters.env;
+    const chainType = ["ethereum", "polygon", "binance"][currentApp.chain];
+    store.dispatch(env + "/updateChainType", chainType);
+    const unlimitedBytes = 10995116277760;
+
+    if (currentApp.storage_limit < unlimitedBytes) {
+      let storage;
+
+      // Convert to MB if storage is less than 1023 MB
+      // else convert to GB
+      if (currentApp.storage_limit <= bytes("1023MB")) {
+        storage = bytes(currentApp.storage_limit, {
+          unitSeparator: " ",
+          unit: "MB",
+        });
+      } else {
+        storage = bytes(currentApp.storage_limit, {
+          unitSeparator: " ",
+          unit: "GB",
+        });
+      }
+
+      const storageValues = storage.split(" ");
+      store.dispatch(env + "/updateStorage", {
+        value: storageValues[0],
+        unit: storageValues[1],
+        isUnlimited: false,
+      });
+    } else {
+      store.dispatch(env + "/updateStorage", {
+        value: "",
+        unit: "",
+        isUnlimited: true,
+      });
+    }
+
+    if (currentApp.bandwidth_limit < unlimitedBytes) {
+      let bandwidth;
+
+      // Convert to MB if bandwidth is less than 1023 MB
+      // else convert to GB
+      if (currentApp.bandwidth_limit <= bytes("1023MB")) {
+        storage = bytes(currentApp.bandwidth_limit, {
+          unitSeparator: " ",
+          unit: "MB",
+        });
+      } else {
+        storage = bytes(currentApp.bandwidth_limit, {
+          unitSeparator: " ",
+          unit: "GB",
+        });
+      }
+
+      const bandwidthValues = bandwidth.split(" ");
+      store.dispatch(env + "/updateBandwidth", {
+        value: bandwidthValues[0],
+        unit: bandwidthValues[1],
+        isUnlimited: false,
+      });
```

Comment:
Looks like this block of code is repeated twice.

Reply:
One is for storage, one is for bandwidth, although both are similar

Reply:
Hmm, we can still abstract it out to a function by passing in the property name...

Reply:
Modified this a bit. Let me know if this works

Reply:
Looks good, @shrinathprabhu.

Reply:
LGTM